zip-0316.rst

```diff
@@ -599,20 +649,144 @@ to give access only to view incoming payments (as opposed to change).
+
+As of `Revision 1`_ of this ZIP, the subset of Metadata Typecodes in
+the range :math:`\mathtt{0xE0}` to :math:`\mathtt{0xFC}` inclusive are
+designated as "MUST-understand": if a Consumer is unable to recognise
+the meaning of a Metadata Item with a Typecode in this range, then it
+MUST regard the entire UA/UVK as unsupported and not process it further.
+
+A `Revision 0`_ UA/UVK (determined by its HRP as specified in
+`Revisions`_) MUST NOT include any Metadata Items with a MUST-understand
+Typecode; a Consumer MUST reject as invalid any UA/UVK that violates
+this requirement.
 
 Since Metadata Items are not Receivers, they MUST NOT be selected by
 a Sender when choosing a Receiver to send to, and since they are not
 Viewing Keys, they MUST NOT provide additional authority to view
 information about transactions.
 
-Currently no Metadata Types are defined. New Metadata Types SHOULD be
-introduced either by a modification to this ZIP or by a new ZIP, in
-accordance with the ZIP Process [#zip-0000]_.
+New Metadata Types SHOULD be introduced either by a modification to this
+ZIP or by a new ZIP, in accordance with the ZIP Process [#zip-0000]_.
+
+Rationale for making Revision 0 UA/UVKs with MUST-understand Typecodes invalid
+''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
 
+.. raw:: html
+
+   <details>
+   <summary>Click to show/hide</summary>
+
+A Consumer implementing this ZIP prior to `Revision 1`_ will not
+recognize the Human-Readable Parts beginning with “``ur``” that mark a
+`Revision 1`_ UA/UVK. So if a UA/UVK that includes MUST-understand
+Typecodes is required to use these `Revision 1`_ HRPs, this will ensure
+that the MUST-understand specification is correctly enforced even for
+such implementations.
+
+.. raw:: html
+
+   </details>
+
+
+Address Expiration Metadata
+---------------------------
+
+As of `Revision 1`_, Typecodes :math:`\mathtt{0xE0}` and :math:`\mathtt{0xE1}`
+are reserved for optional address expiry metadata. A producer MAY choose to
+generate Unified Addresses containing either or both of the following Metadata
+Item Types, or none.
+
+The value of a :math:`\mathtt{0xE0}` item MUST be an unsigned 32-bit integer in
+little-endian order specifying the Address Expiry Height, a block height of the
+Zcash chain associated with the UA/UVK. A Unified Address containing metadata
+Typecode :math:`\mathtt{0xE0}` MUST be considered expired when the height of
+the Zcash chain is greater than this value.
+
+The value of a :math:`\mathtt{0xE1}` item MUST be an unsigned 64-bit integer in
+little-endian order specifying a Unix Epoch Time, hereafter referred to as the
+Address Expiry Time. A Unified Address containing Metadata Typecode
+:math:`\mathtt{0xE1}` MUST be considered expired when the current time is
+after the Address Expiry Time.
+
+A Sender that supports `Revision 1`_ of this specification MUST set
+a non-zero ``nExpiryHeight`` field in transactions that it creates. If the
```

Comment:
```suggestion
a non-zero ``nExpiryHeight`` field in transactions that it creates if they 
are sending to an address that defines an Address Expiry Height. If the
```

Reply:
(I changed "they are" to "it is" because a Sender is a process acting on behalf of a user, not the user themself. We use this convention generally in Zcash docs.)